### PR TITLE
DOP-922: Report diagnostics with a separate return code

### PR DESCRIPTION
**Pull request by i80and**
* Return status code 2 if there are build errors
* Status code 1 now indicates a crash
* Fixed up and refactored our diagnostic levels

### CODE REVIEW

File: snooty/main.py [hunk 1]
```diff
@@ -275,4 +282,4 @@ def main() -> None:
             connection.close()
 
     if args["build"] and backend.total_warnings > 0:
```
Comment (skerschb) [inline snooty/main.py:284]: Do we want to make this more of a threshold?

File: snooty/main.py [hunk 2]
```diff
@@ -284,5 +288,5 @@ def main() -> None:
             print("Closing connection...")
             connection.close()
 
-    if args["build"] and backend.total_warnings > 0:
-        sys.exit(1)
+    if args["build"] and backend.total_errors > 0:
+        sys.exit(EXIT_STATUS_ERROR_DIAGNOSTICS)
```
Comment (madelinezec) [inline snooty/main.py:292]: So what is the difference between exit code 1 and 2 here? How does returning exit code 2 make it so the autobuilder can catch unexpected parser terminations?

Comment (i80and) [reply snooty/main.py:292]: Exit code 1 indicates unexpected termination, while 2 indicates that the parse finished, but with errors the writer should definitely look at. If error code 1 is returned, then the autobuilder should not run the frontend at all.

File: snooty/diagnostics.py [hunk 3]
```diff
@@ -29,15 +29,14 @@ def __init__(
         self.end = (end_line, end_column)
 
     class Level(enum.IntEnum):
-        info = 0
-        error = 1
+        info = 1
         warning = 2
+        error = 3
```
Comment (mango-db) [inline snooty/diagnostics.py:34]: Is there a reason why the internal `Level` numbering is the opposite of standard (`error=1`, etc...)?

Comment (i80and) [reply snooty/diagnostics.py:34]: This matches the internal doctutils diagnostic code scheme, but that's mostly an arbitrary decision! It is however more common for increasingly severe error conditions to have higher values (see: Python, Java standard libraries). Language Server Protocol is the odd one out, as far as I'm aware.

**Pull request merged**